harness/determined/_train_context.py

```diff
@@ -18,6 +18,14 @@ def __init__(self, env: det.EnvContext, hvd_config: horovod.HorovodContext):
         self.distributed = DistributedContext(env, hvd_config)
         self._stop_requested = False
 
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> "_TrainContext":
+        env_context, rendezvous_info, hvd_config = det._make_local_execution_env(
+            training=False,
```

Comment:
it looks this feature is used for training. Why is it set to be false?

Reply:
because `training` really means "training inside determined", and was originally contrasted with "exporting" basically.

In this case, users are using their trial code outside of determined and we should just hand them back their unmodified models and optimizers and such, as with checkpoint export.

Reply:
fun fact, after writing this code I found `det.experimental.create_trial_instance` makes the same exact calls, only it inverts the control that the user would have compared to `_TrainContext.from_config()`.

Reply:
This brings up a good point though.  We currently have this experimental function that we've never promoted (and which I had totally forgotten about) called `create_trial_instance()`.  The behavior is very similar, but the API has bit of a different feel to it:

```python
# 1. Proposed API
config = { ... }
context = det.pytorch.PyTorchTrialContext.from_config(config)
trial = MyPyTorchTrial(context)
```

vs

```python
# 2. API similar to what we already have
config = { ... }
trial = det.create_trial_instance(MyPyTorchTrial, config)
```

`1` is a bit more verbose but perhaps a bit less magical, since users would actually be instantiating their Trial manually rather than calling a function that does the instantiation automatically.  Not sure what real users would think of that tradeoff, or if it would even matter very much.

@aaron276h @shiyuann @davidhershey, what do you think?

Reply:
I guess there is also a third option:
```python
# 3. Just an idea
config = { ... }
trial = MyPyTorchTrial.from_config(config)
```
But I think it might be a bit confusing to users how we go from a `config` in `MyPyTorchTrial.from_config(config)` to a `context` in `MyPytorchTrial.__init__(context)`

Also, if you allow bare `context` objects via `det.pytorch.PyTorchTrialContext.from_config(config)` you enable users to interactively play with e.g. data layer functions in a notebook without having to go through a Trial class.

Yeah, actually I think I really option `1` because I like exposing the context object and letting users feed it to their own trial.

Reply:
I think option 1 is definitely the best one, its the most clear about what is going on and the extra line users will be required to add isn't a big overhead.

Reply:
I also think 1) is the best option.